### PR TITLE
Added project status PNG snapshot on project status report publish

### LINKED ISSUE #324: Ability to export project status report as snapshot/image/PDF
**Is your feature request related to a problem? Please describe.**
Project snapshotting is useful for persisting the full project status page, including list data. We should look into how this can be solved.

**Describe the solution you'd like**
When a report is published, we could also snapshot the report (as PDF or image) and save it as an attachment to the published status report.

### PR DESCRIPTION
**Pull request by Petterandre**
### Your checklist for this pull request

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your main!
- [x] Make sure you are making a pull request against the **dev** branch (left side). Also you should start *your branch* off *dev*.
- [x] Check the commit's or even all commits' message 
- [x] Check your code additions will fail linting checks
- [x] Remember: Add PR description to [Changelog](https://github.com/Puzzlepart/prosjektportalen365/blob/dev/CHANGELOG.md) with the ID that matches this PR

### Description

When publishing a status report, adds a snapshot available on the project status page.
The file is attached to the list element.

PDF is not included, as it proves much more challenging. Will be on the back burner

### How to test

- [x] 1. Create a new status report
- [x] 2. Publish
- [x] 3. View snapshot

Closes #324 

💔Thank you!

### CODE REVIEW

File: SharePointFramework/@Shared/src/services/PortalDataService/index.ts [hunk 1]
```diff
@@ -54,14 +57,17 @@ export class PortalDataService {
...
     /**
      * Update status report
      * 
      * @param {number} id Id
      * @param {TypedHash<string>} properties Properties
      */
-    public async updateStatusReport(id: number, properties: TypedHash<string>): Promise<void> {
+    public async updateStatusReport(id: number, properties: TypedHash<string>, attachment: {fileName: string, content: Blob}): Promise<void> {
+        await this._web.lists.getByTitle(this._configuration.listNames.PROJECT_STATUS).items.getById(id).attachmentFiles.add(attachment.fileName, attachment.content);
         await this._web.lists.getByTitle(this._configuration.listNames.PROJECT_STATUS).items.getById(id).update(properties);
```
Comment (olemp) [inline SharePointFramework/@Shared/src/services/PortalDataService/index.ts:69]: Initialize a variable with the list first? More readable.

File: SharePointFramework/ProjectWebParts/src/components/ProjectStatus/index.tsx [hunk 2]
```diff
@@ -139,12 +141,21 @@ export class ProjectStatus extends React.Component<IProjectStatusProps, IProject
         href: sourceUrl,
       })
     }
+    farItems.push({
+      id: "GetSnapshot",
+      key: "GetSnapShot",
+      name: "Åpne som øyeblikksbilde",
```
Comment (olemp) [inline SharePointFramework/ProjectWebParts/src/components/ProjectStatus/index.tsx:147]: Add resources to `strings`

**Review by olemp — CHANGES_REQUESTED** [review]
... also, how is errors handled? If there's an error attaching  the PNG for instance? Is it graceful?

Also, resolve conflicts and run `npm run lint`.

(I hate this feature so much)

**Comment by Petterandre** [comment]
> ... also, how is errors handled? If there's an error attaching the PNG for instance? Is it graceful?
> 
> Also, resolve conflicts and run `npm run lint`.
> 
> (I hate this feature so much)

Errors now handled. lint script does not exist in @Shared, but should pass, looking at the existing code.

I'm sorry, I know you hate it, dear.

**Comment by Petterandre** [comment]
There are a few more fixes incoming.

**Comment by olemp** [comment]
> There are a few more fixes incoming.

Use draft mode then 🚀

**Comment by olemp** [comment]
Run `lint` please 👨‍🎤 🎸

**Comment by Petterandre** [comment]
Ran lint once more, there was an unlinted file. Had to remove prettier temporarily, as it failed for some unknown reason.

**Comment by pzljanb** [comment]
Works well

**Pull request merged**